src/Lecture2.hs

```diff
@@ -150,14 +206,62 @@ You're free to define any helper functions.
        treasure besides gold (if you already haven't done this).
 -}
 
--- some help in the beginning ;)
+newtype Health = Health Int deriving Show
+newtype Attack = Attack Int deriving Show
+
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
+    { knightHealth    :: Health
+    , knightAttack    :: Attack
     , knightEndurance :: Int
     }
 
-dragonFight = error "TODO"
+data Color = Red | Black | Green
+
+data Dragon = Dragon
+    { dragonColor  :: Color
+    , dragonHealth :: Health
+    , dragonAttack :: Attack
+    }
+
+newtype Experience = Experience Int deriving Show
+newtype Gold = Gold Int deriving Show
+
+data Reward
+    = WithTreasure Experience Gold
+    | WithoutTreasure Experience Gold
```

Comment:
I clarified the task a bit to cover cases so dragons would determine the treasure itself, not just "yes" or "no". It will make the implementation a bit more challenging.

But, besides that, the current implementation covers the requirements nicely 👍🏻